pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -536,6 +536,9 @@ type Template struct {
 
 	// Memoize allows templates to use outputs generated from already executed templates
 	Memoize *Memoize `json:"memoize,omitempty" protobuf:"bytes,37,opt,name=memoize"`
+
+	// ResourcesDuration is the total used by the template
+	ResourcesDuration ResourcesDuration `json:"resourcesDuration,omitempty" protobuf:"bytes,38,opt,name=resourcesDuration"`
```

Comment:
This is part of the spec field, not the status field. Fields here are filled in by the user as specifications, not by argo as status report. This field does not belong here.